bin/utest/exec.c

```diff
@@ -0,0 +1,45 @@
+#include "utest.h"
+
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/wait.h>
+#include <assert.h>
+
+int test_execve() {
+  int pid = fork();
+  if (pid) {
+    /* parent */
+    int status;
+    int r = wait(&status);
+    assert(r == pid);
```

Comment:
It is not clear for me how are you catching `fork` failures here.

Let's assume that `fork` is broken. Then `pid` is -1. `wait` should failed and `r` will be -1. `status` is uninitialized here so it can be equal to 0. Both assertion will pass and you return 0 which means test success but you never call `execve`.

Reply:
Thanks. I've added error handling.